src/eda_script.R

```diff
@@ -14,25 +14,45 @@ Options:
 
 library(docopt)
 library(tidyverse)
-library(GGally)
+library(reshape2)
 library(gridExtra)
 
 main <- function(input, output) {
   
   # test to check that csv is inputted 
   if (substr(input, (nchar(input)+1)-3 ,nchar(input)) != "csv"){
-    stop("Must input an Rmd file")
+    stop("Must input an csv file")
   }
   training_data  <- read_csv(input)
   
   # Correlation plot
-  corr_plot  <- training_data  %>%
-    select(-neighbourhood_group, -neighbourhood, -room_type, -latitude, -longitude )  %>%
-    rename(no_host_listgs = calculated_host_listings_count )  %>%
-    ggpairs() +
-    theme(strip.text.x = element_text(size = 8),
-          strip.text.y = element_text(size = 8),
-          axis.text.x = element_text(angle=90))
+  cors  <- training_data  %>%
+    select(-neighbourhood_group, -neighbourhood, -room_type, -latitude, -longitude )  %>% 
+    cor()
+  
+  get_lower_tri<-function(cormat){
+    cormat[upper.tri(cormat)] <- NA
+    return(cormat)
+  }
+  # Get upper triangle of the correlation matrix
+  get_upper_tri <- function(cormat){
+    cormat[lower.tri(cormat)]<- NA
+    return(cormat)
+  }
+  
+  upper_tri  <- get_upper_tri(cors)
+  cors  <- melted_cormat <- melt(upper_tri, na.rm = TRUE)
+  
+  corr_plot  <- ggplot(data = cors, aes(Var2, Var1, fill = value))+
```

Comment:
Good job on the correlation plot! 

<img width="954" alt="Screen Shot 2020-02-07 at 6 14 07 PM" src="https://user-images.githubusercontent.com/25375401/74077583-ad258380-49d5-11ea-80a7-ae638278fe94.png">

The current scale is a bit confusing though. Any chance that you could use the same colour scale so that it's easier to compare? For instance, a smaller correlation would just be a lighter red as opposed to purple.

Reply:
The way the plot is set up, it has a color for high, mid and low correlation. So making them all a different shade of red makes it even more confusing. The way it currently is, it's not as confusing, as the user just needs to look at the legend to the right. You can see that a light purple shade is a small negative number.

Reply:
My bad. Didn't notice that purple is for negative numbers. Thank god you didn't listen to my BS haha. All good!